parser/openfoodfacts_taxonomy_parser/parser/parser.py

```diff
@@ -1,158 +1,213 @@
-            FOR (n:{project_name}) ON EACH [n.id]\n"""
-            + """
-            OPTIONS {indexConfig: {`fulltext.analyzer`: 'keyword'}}"""
+            if child_link["parent_id"] in node_ids:
+                normalised_child_links.append(child_link)
+            else:
+                unnormalised_child_links.append(child_link)
+
+        normalised_query = f"""
+            UNWIND $normalised_child_links as child_link
+            MATCH (p:{project_label}) USING INDEX p:{project_label}(id)
+            WHERE p.id = child_link.parent_id
+            MATCH (c:{project_label}) USING INDEX c:{project_label}(id)
+            WHERE c.id = child_link.id
+            CREATE (c)-[relations:is_child_of]->(p)
+            WITH relations
+            UNWIND relations AS relation
+            RETURN COUNT(relation)
+        """
+
+        language_codes = set()
+        for child_link in unnormalised_child_links:
+            lc, parent_id = child_link["parent_id"].split(":")
+            language_codes.add(lc)
+            child_link["parent_id"] = parent_id
+
+        parent_id_query = " OR ".join(
+            [f"child_link.parent_id IN p.tags_ids_{lc}" for lc in language_codes]
         )
-        self.session.run(query)
 
-        language_codes = [lang.alpha2 for lang in list(iso639.languages) if lang.alpha2 != ""]
-        tags_prefixed_lc = ["n.tags_" + lc for lc in language_codes]
-        tags_prefixed_lc = ", ".join(tags_prefixed_lc)
-        query = f"""CREATE FULLTEXT INDEX {project_name+'_SearchTags'} IF NOT EXISTS
-            FOR (n:{project_name}) ON EACH [{tags_prefixed_lc}]"""
-        self.session.run(query)
+        unnormalised_query = f"""
+            UNWIND $unnormalised_child_links as child_link
+            MATCH (p:{project_label})
+            WHERE {parent_id_query}
+            MATCH (c:{project_label}) USING INDEX c:{project_label}(id)
+            WHERE c.id = child_link.id
+            CREATE (c)-[relations:is_child_of]->(p)
+            WITH relations
+            UNWIND relations AS relation
+            RETURN COUNT(relation)
+        """
+        count = 0
+
+        if normalised_child_links:
+            normalised_result = self.session.run(
+                normalised_query, normalised_child_links=normalised_child_links
+            )
+            count += normalised_result.value()[0]
+
+        if unnormalised_child_links:
+            unnormalised_result = self.session.run(
+                unnormalised_query, unnormalised_child_links=unnormalised_child_links
+            )
+            count += unnormalised_result.value()[0]
+
+        self.parser_logger.info(
+            f"Created {count} 'is_child_of' links in {timeit.default_timer() - start_time} seconds"
+        )
+
+        if count != len(child_links):
+            self.parser_logger.error(
+                f"Created {count} 'is_child_of' links, {len(child_links)} links expected"
+            )
```

Comment:
This is not a show stopper but we could add it to error nodes.

Reply:
This is already the case (`parser_logger.error` append errors to a list which we then add to the error nodes)

Reply:
oups, that's true ;-)